### DISCUSSION TITLE
System.Invalid Operation Exception

### DISCUSSION THREAD

**Discussion by Amorganskate** [ZiggyCreatures/FusionCache#282] [General]
We just started implementing this fusion cache library and noticed some exceptions in our logs that happen pretty infrequently but wanted to have a discussion about it and see if its just from our logger levels or if we have something going on that we need to resolve.

```
fail: ZiggyCreatures.Caching.Fusion.FusionCache[0]
      FUSION [N=FusionCache I=cfbac876cf4841a695a80951ffd645fb] (O=0HN5N0I89EEKC K=***** a background factory thrown an exception
System.InvalidOperationException: Invalid operation. The connection is closed.
   at System.Threading.Tasks.ContinuationResultTaskFromResultTask`2.InnerInvoke()
   at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state)

```

**Comment by jodydonetti** [comment]
Hi @Amorganskate and thanks for using FusionCache!

I suppose you are using [soft/hard timeouts](https://github.com/ZiggyCreatures/FusionCache/blob/main/docs/Timeouts.md) or [eager refresh](https://github.com/ZiggyCreatures/FusionCache/blob/main/docs/EagerRefresh.md), right?

These features may let a factory run (or finish running) in the background: this means that if you close the connection after a `GetOrSet`, but before the background factory finished running, you are effectively closing a connection _while_ it is being used.

Usually this happens with EF, and you can see [here](https://github.com/ZiggyCreatures/FusionCache/issues/21) and [here](https://github.com/ZiggyCreatures/FusionCache/issues/119) other community members stumbling into this (and the solution they've used).

I also opened an issue on the EF repo to start a discussion about potential solutions, which you can see [here](https://github.com/dotnet/efcore/issues/33892).

Hope this helps!

**Reply by Amorganskate** [reply]
You're the man Jody, my team and I appreciate the speedy reply. We are going to take some time and get one of these implementations up and going. We will let you know the results!
